File: 242.valid-anagram.cs
/*
 * @lc app=leetcode id=242 lang=csharp
 *
 * [242] Valid Anagram
 */

// @lc code=start
public class Solution {
    public bool IsAnagram(string s, string t) {
        if(s.Length != t.Length) return false;
        int[] count = new int[26];
        foreach(var c in s) count[c - 'a']++;
        foreach(var c in t) if(count[c - 'a']-- == 0) return false;
        return true;
    }
}
// @lc code=end

File: 242.valid-anagram.js
/*
 * @lc app=leetcode id=242 lang=javascript
 *
 * [242] Valid Anagram
 */

// @lc code=start
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function (s, t) {
  if (s.length != t.length) {
    return false;
  }
  var array = [];
  for (var c in s) {
    var i = s.charCodeAt(c) - 97;
    array[i] = (array[i] || 0) + 1;
  }
  for (var c in t) {
    var i = t.charCodeAt(c) - 97;
    if (!array[i]) return false;
    array[i]--;
  }
  return true;
};
// @lc code=end
